includes/Core/Modules/Modules.php

```diff
@@ -449,17 +449,17 @@ private function get_rest_routes() {
 				)
 			),
 			new REST_Route(
-				'modules/(?P<slug>[a-z\-]+)',
+				'core/modules/data/(?P<slug>[a-z\\-]+)',
```

Comment:
Can you clarify what you mean when you say the JS API / `apiFetch` don't support `data` parameters? In case of `GET` requests, any `data` should simply be passed as query parameters. Is that not the case? I was a little surprised by that.

Generally, putting the slug as datapoint would be a nice idea if we were using a more regular REST API naming scheme, but with where we are it's a bit strange because right now `list` takes kind of the same "meaning" as a module slug.

Reply:
> Can you clarify what you mean when you say the JS API / apiFetch don't support data parameters? In case of GET requests, any data should simply be passed as query parameters. Is that not the case? I was a little surprised by that.

Yes, that's what I thought would happen too but the request fails with a `fetch_error`:
![image](https://user-images.githubusercontent.com/1621608/76026034-b3f0d900-5f36-11ea-8b98-557f91541dd0.png)

Since `apiFetch` is just a wrapper around `fetch`, [the docs](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch) do say that it can't have a `body` which is what `data` translates to.
[![image](https://user-images.githubusercontent.com/1621608/76026147-e13d8700-5f36-11ea-82c5-ec1786b2dd7a.png)](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch)

That doesn't mean we couldn't implement this in `dataAPI.get` if we wanted to. I'd prefer to keep the slug as a route parameter if we could though.

> Generally, putting the slug as datapoint would be a nice idea if we were using a more regular REST API naming scheme, but with where we are it's a bit strange because right now list takes kind of the same "meaning" as a module slug.

How about we keep it under modules as `modules/{slug}/data/activation` ?

Reply:
> Sure, you can't have a body in a GET request, but both our current API and new API shouldn't do that

Ah, yeah I was testing with `apiFetch` directly. For some reason I thought our API layer didn't do this. Not a problem then! 👍 

> We generally separate between the "core parts" of dealing with general module tasks (including knowing which modules even exist) and the "module parts" (specific functionality per module), and currently the differentiation on a REST API level is that core/modules vs modules/{slug} prefix. We would blur this line by making the change you're suggesting.

I don't disagree with that separation, I just think you could put the endpoints for individual module activation and info under a module-specific route since the request is by definition module-specific. The fact that a single valid slug is required is what makes the differentiation clear in my eyes.

Similarly, I think using the same route for getting module info for a single module, and POSTing to the same route to activate/deactivate the module is somewhat unintuitive.

I realize the design we're building around is a bit different from what might be more standard REST routes, so that's probably why this feels a bit strange to me.

I'll go ahead and implement it as `core/modules/data/info`; just sharing my thoughts as I didn't elaborate on the reasoning before.

Reply:
@aaemnnosttv 
> I don't disagree with that separation, I just think you could put the endpoints for individual module activation and info under a module-specific route since the request is by definition module-specific. The fact that a single valid slug is required is what makes the differentiation clear in my eyes.

The main reason why I think we should keep the generic info and activation parts as part of `core/modules` is that this functionality should be available regardless of whether the respective module is active or not. Anything on `modules/{slug}` is module-specific and should typically only exist if the module is active. Same on the JS side, where module-specific datastores are only relevant if that module is active. The generic info to display for a module and whether it's active or not must still be available regardless of that.

However, I like your idea of separating `core/modules/data/info` from the activation part. How about we put the `POST` part of that to `core/modules/data/activation` or something similar? We can also simplify the expected parameter and return value for that one maybe, it doesn't necessarily need to return the entire module information.

Reply:
> The main reason why I think we should keep the generic info and activation parts as part of `core/modules` is that this functionality should be available regardless of whether the respective module is active or not. Anything on `modules/{slug}` is module-specific and should typically only exist if the module is active

In this case, shouldn't modules register their own `modules/{slug}` routes? We could add a new `Module_With_REST_Routes` interface+trait. This would also simplify the callback for all of the module's endpoints, because the module instance wouldn't have to be looked up and validated. This warrants its own issue of course if you agree, but let me know what you think.

> However, I like your idea of separating `core/modules/data/info` from the activation part. How about we put the `POST` part of that to `core/modules/data/activation` or something similar? We can also simplify the expected parameter and return value for that one maybe, it doesn't necessarily need to return the entire module information.

Cool, I'll do that 👍